src/main/java/com/eleks/academy/whoami/core/state/SuggestingCharacters.java

```diff
@@ -59,98 +49,90 @@ public GameStatus getStatus() {
+				break;
+			}
+		} while (countShuffledCharacters != FAILED_ATTEMPTS_SHUFFLED);
+
+		if (countShuffledCharacters == FAILED_ATTEMPTS_SHUFFLED) {
+			playerCharacterShuffled = moveCharacters(playerToCharacterCopy,
+					new Random().nextInt(playerToCharacterCopy.size() - 1) + 1);
 		}
 
-		characters.add(GameCharacter.of(character, player));
-
-		return this;
-	}
+		fillPlayersWithShuffledCharacters(this.players, playerCharacterShuffled);
 
-	@Override
-	public Map<String, SynchronousPlayer> getPlayers() {
 		return this.players;
 	}
 
-	/**
-	 * The term author is referred to a player who suggested at least one character
-	 * <p>
-	 * Basic algorithm description:
-	 * 1) Collect all randomly-ordered authors into a form of cyclic oriented graph.
-	 * 2) Assign to each author a random character suggested by the next author (next graph node)
-	 * 3) Randomly assign all the suggested characters that are left disregarding the author to
-	 * all the non-author players
-	 */
-	private GameState assignCharacters() {
-		Function<String, Integer> randomAuthorOrderComparator = value ->
-				Double.valueOf(Math.random() * 1000).intValue();
+	private Map<String, String> shuffledCharacters(Map<String, String> playerCharacter) {
+		List<String> key = new ArrayList<>(playerCharacter.keySet());
+		List<String> value = new ArrayList<>(playerCharacter.values());
 
-		final var authors =
-				this.suggestedCharacters.keySet()
-						.stream()
-						.sorted(Comparator.comparing(randomAuthorOrderComparator))
-						.collect(Collectors.toList());
+		Collections.shuffle(value);
 
-		authors.forEach(author -> {
-			final var character = this.getRandomCharacter()
-					.apply(this.suggestedCharacters.get(this.<String>cyclicNext().apply(authors, author)));
+		playerCharacter = IntStream.range(0, playerCharacter.size()).boxed()
+				.collect(Collectors.toMap(key::get, value::get));
 
-			character.markTaken();
+		return playerCharacter;
+	}
 
-			this.playerCharacterMap.put(author, character.getCharacter());
-		});
+	private boolean atLeastOneEqual(Map<String, String> oldPlayerCharacter, Map<String, String> playerCharacterShuffled) {
+		int countEquals = 0;
 
-		final var authorsSet = new HashSet<>(authors);
+		for (Map.Entry<String, String> entry : oldPlayerCharacter.entrySet()) {
+			String key = entry.getKey();
+			String val1 = entry.getValue();
+			String val2 = playerCharacterShuffled.get(key);
+			boolean isTwoValuesEqual = val1.equals(val2);
 
-		final var nonTakenCharacters = this.suggestedCharacters.values()
-				.stream()
-				.flatMap(Collection::stream)
-				.filter(character -> !character.isTaken())
-				.collect(toList());
+			if (isTwoValuesEqual) {
+				countEquals++;
+			}
+		}
 
-		this.players.keySet()
-				.stream()
-				.filter(player -> !authorsSet.contains(player))
-				.forEach(player -> {
-					final var character = this.getRandomCharacter().apply(nonTakenCharacters);
+		return countEquals <= 0;
```

Comment:
if `int countEquals = 0;` and in the end `countEquals <= 0` then better to compare `countEquals == 0`

in your case method better would sound like `areNotEqual`.

If right now it works then update the condition and update the method name

Reply:
Resolved